File: leetcode/@DataStructure/217.py
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        dictionary = {}
        for number in nums:
            if dictionary.get(number) != 1:
                dictionary[number] = 1
            else:
                return True
        return False


solotion = Solution()
r = solotion.containsDuplicate([1, 2, 3, 1])
print(r)

File: crack_coding_interview/c1/217.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
const containsDuplicate = (nums) => {
  const map = {};
  let result = false;
  nums.forEach((num) => {
    if (map[num]) {
      result = true;
    } else {
      map[num] = true;
    }
  });
  return result;
};
